### ISSUE TITLE
Die, if Read 1 and Read 2 have different number of reads

### ISSUE THREAD

**Issue by FelixKrueger** [FelixKrueger/Bismark#219]
This can be either a result of file truncation, or simply that R1 and R2 were specified incorrectly (e.g. one trimmed and one untrimmed file).

**Comment by FelixKrueger** [comment]
Added a check for the number of reads in R1 and R2. This is implemented once the in-silico bisulfite transformation of reads has completed. 

Until now, e.g. Bowtie2 would crash mid way, but this is hard to spot:

```
Error, fewer reads in file specified with -2 than in file specified with -1
terminate called after throwing an instance of 'int'
(ERR): bowtie2-align died with signal 6 (ABRT) 
```

Now Bismark terminates with a `[FATAL ERROR]` before the alignment process is started, this should save a lot of headaches. Fixed in 1ff5cd484c7e56c0453fafb50d9734b563b0ac3a.

**Comment by FelixKrueger** [comment]
Just to add, this works in `directional`, `--pbat` and `--non_directional` mode for FastQ files only. Furthermore, it works for paired-end mode with Bowtie 2, and Bowtie 1 as long as the temporary files are not to be written out in a compressed form.
